server-spi-private/src/main/java/org/keycloak/policy/AgePasswordPolicyProvider.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.policy;
+
+import org.keycloak.common.util.Time;
+import org.keycloak.credential.hash.PasswordHashProvider;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.PasswordPolicy;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.models.credential.PasswordCredentialModel;
+import org.jboss.logging.Logger;
+
+import java.time.Duration;
+
+/**
+ * @author <a href="mailto:dev.maciej.mierzwa@gmail.com">Maciej Mierzwa</a>
+ */
+public class AgePasswordPolicyProvider implements PasswordPolicyProvider {
+    private static final String ERROR_MESSAGE = "invalidPasswordGenericMessage";
+    public static final Logger logger = Logger.getLogger(AgePasswordPolicyProvider.class);
+    private final KeycloakSession session;
+
+    public AgePasswordPolicyProvider(KeycloakSession session) {
+        this.session = session;
+    }
+
+    @Override
+    public PolicyError validate(String user, String password) {
+        return null;
```

Comment:
It looks like this method is ultimately called only in tests, but wouldn't it be better to invoke the other variant from here instead of returning `null`? Something like

```
RealmModel realm = session.getContext().getRealm();
return validate(realm, session.users().getUserByUsername(realm, user), password);
```
Just a suggestion, I wouldn't reject the PR because of this.

Reply:
Def agreed, pushed changes with your suggestion